morph/react/block-add-data.js

```diff
@@ -3,20 +3,44 @@ import toCamelCase from 'to-camel-case'
 import path from 'path'
 
 export function enter(node, parent, state) {
-  for (let data of node.data) {
-    data.name = getDataVariableName(data, state)
-
-    // at the moment it will create multiple instances for the same data key
-    // an optimization will be implemented to reuse a variable if possible
-    state.dataBlocks.push(
-      `let ${data.name} = fromData.useData({ viewPath, path: '${data.path}', `
-    )
-    maybeDataContext(data, state)
-    maybeDataFormat(data.format, state)
-    maybeDataValidate(data.validate, state)
-    state.dataBlocks.push('})')
-
-    state.use('ViewsUseData')
+  for (let aggregateData of node.data) {
+    for (let data of aggregateData.data) {
+      data.name = getDataVariableName(data, state)
+
+      // at the moment it will create multiple instances for the same data key
+      // an optimization will be implemented to reuse a variable if possible
+      state.dataBlocks.push(
+        `let ${data.name} = fromData.useData({ viewPath, path: '${data.path}', `
+      )
+      maybeDataContext(data, state)
+      maybeDataFormat(data.format, state)
+      maybeDataValidate(data.validate, state)
+      state.dataBlocks.push('})')
+
+      state.use('ViewsUseData')
+    }
+
+    if (aggregateData.aggregate) {
+      aggregateData.name = getAggregateDataVariableName(state)
+      let importName = getAggregateImportName(
+        aggregateData.aggregate.source,
+        state
+      )
+      state.dataBlocks.push(
+        `let ${aggregateData.name} = React.useMemo(() => ${importName}.${
```

Comment:
What do you think... Do we want to try first without memo or do you already see those being heavy enough to have it in to begin with?

Reply:
not sure, probably the aggregate functions should be fast enough. I could get rid of the useMemo for now and we can reevaluate if it is needed later

Reply:
cool, maybe leave the code that generates the memo commented out so we have it handy for later just in case?

Reply:
do you think this is necessary to have the useMemo commented out? the logic is quite simple so I think we can easily put it back when needed

Reply:
entirely up to you :D